### PR TITLE
Separates robot_items.dm and cleans up the code

### PR DESCRIPTION
**Pull request by SuperNovaa41**
I want to do a refactor for an item here, but I'd rather split all of
this up first, so as not to have a boat load of changes to review all
together. as well this is all five year old code and frankly kinda sucks

WHY ITS GOOD

good code, cleans up some ancient messes, documents a lot of old variables and procs

:cl:
fix: Fixes an issue with service borg apparatuses getting multiple beakers in it's contents.
/:cl:

### CODE REVIEW

**Comment by SuperNovaa41** [comment]
Probably going to do some more work on the code but i've got things to do, and figured i'd put it up for review

**Comment by ATH1909** [comment]
> fix: Fixes an issue with service borg apparatuses getting multiple beakers in it's contents.

could you elaborate on this?

**Comment by SuperNovaa41** [comment]
> > fix: Fixes an issue with service borg apparatuses getting multiple beakers in it's contents.
> 
> could you elaborate on this?

the borg beaker apparatus creates a beaker on init, and then sets `stored` to equal that.
the service borg has a subtype of that which creates a glass and sets it to `stored`, however the other beaker is still stored in its contents (its just inaccessible)

File: code/game/objects/items/robot/items/food.dm [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+	if(candy <= 0)
+		to_chat(user, span_warning("No treats left in storage!"))
+		return FALSE
+	var/turf/turf = get_turf(atom)
+	if(!turf || !istype(turf) || !isopenturf(turf))
+		return FALSE
+	if(isobj(atom))
+		var/obj/obj = atom
+		if(obj.density)
+			return FALSE
```
Comment (MrMelbert) [inline code/game/objects/items/robot/items/food.dm:56]: `turf` is no more descriptive var name than `t`, same with `atom` and `obj`

File: code/game/objects/items/robot/items/food.dm [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+		if(DISPENSE_LOLLIPOP_MODE)
+			food_item = new /obj/item/food/lollipop/cyborg(turf)
+		if(DISPENSE_ICECREAM_MODE)
+			food_item = new /obj/item/food/icecream(turf, list(ICE_CREAM_VANILLA))
+			food_item.desc = "Eat the ice cream."
+
+	var/into_hands = FALSE
+	if(ismob(atom))
+		var/mob/mob = atom
+		into_hands = mob.put_in_hands(food_item)
```
Comment (MrMelbert) [inline code/game/objects/items/robot/items/food.dm:69]: Same here

File: code/game/objects/items/robot/items/generic.dm [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+
+/obj/item/borg/stun
+	name = "electrically-charged arm"
+	icon_state = "elecarm"
+	/// Cost to use the stun arm
+	var/charge_cost = 1000
+
+/obj/item/borg/stun/attack(mob/living/mob, mob/living/user)
+	if(ishuman(mob))
+		var/mob/living/carbon/human/human = mob
```
Comment (MrMelbert) [inline code/game/objects/items/robot/items/generic.dm:23]: Same here

**Review by MrMelbert — CHANGES_REQUESTED** [review]
double check your var names I only saw these but there may be more

**Pull request merged**